cmd/osctl/cmd/cluster.go

```diff
@@ -100,15 +99,32 @@ func create() (err error) {
 		return err
 	}
 
-	// Generate all PKI and tokens required by Talos.
+	// Validate CIDR range and allocate IPs
+	fmt.Println("validating CIDR and reserving master IPs")
 
-	fmt.Println("generating PKI and tokens")
+	masterIP, _, err := net.ParseCIDR(networkCIDR)
+	if err != nil {
+		return fmt.Errorf("error validating cidr block: %w", err)
+	}
+
+	// Set starting ip at 2nd ip in range, ex: 192.168.0.2
+	masterIP = masterIP.To4()
+	masterIP[3] += 2
 
 	ips := make([]string, masters)
 	for i := range ips {
-		ips[i] = fmt.Sprintf(baseNetwork, i+2)
+		ips[i] = masterIP.String()
+
+		// Make sure we don't overflow b/c some jabroni has requested a hilarious amount of masters
+		if masterIP[3] == 254 {
+			return errors.New("can't allocate any new IPs out of CIDR")
+		}
+		masterIP[3]++
```

Comment:
worker nodes are going to consume IPs as well, are we going to leave it up to docker to resolve?

Reply:
Yep, that was my thought. That was the behavior we had previously.

Reply:
We can remove this if we use  https://github.com/talos-systems/talos/blob/master/pkg/net/net.go#L52.